packages/module-context/src/client/ContextClient.ts

```diff
@@ -0,0 +1,57 @@
+import { Observable, BehaviorSubject, EMPTY } from 'rxjs';
+import { catchError, map } from 'rxjs/operators';
+
+import { Query, QueryCtorOptions } from '@equinor/fusion-observable/query';
+import { QueryClientStatus } from '@equinor/fusion-observable/query/client';
+
+import { ContextItem } from '../types';
+
+export type ContextClientOptions = {
+    query: QueryCtorOptions<ContextItem, { id: string }>;
+    initial?: ContextItem;
+};
+
+export class ContextClient extends Observable<ContextItem> {
+    #client: Query<ContextItem, { id: string }>;
+    /** might change to reactive state, for comparing state with reducer */
+    #currentContext$: BehaviorSubject<ContextItem | undefined>;
+
+    get currentContext(): ContextItem | undefined {
+        return this.#currentContext$.value;
+    }
+
+    get currentContext$(): Observable<ContextItem | undefined> {
+        return this.#currentContext$.asObservable();
+    }
+
+    get status$(): Observable<QueryClientStatus> {
+        return this.#client.client.pipe(map((x) => x.status));
+    }
+
+    get client(): Query<ContextItem, { id: string }> {
+        return this.#client;
+    }
+
+    constructor(options: ContextClientOptions) {
+        super((observer) => this.#currentContext$.subscribe(observer));
```

Comment:
I don`t see any teardown or unsubscribe on this. Do you otherwise complete the currentcontext stream anywhere?
Module is pretty global tho so you can say that the stream lifecycle equals the js instance lifecycle

Reply:
the `super` function exposes the callback when observers observer the instance if `ContextClient`.

`this.#currentContext$.subscribe(observer)` will return a subscription when subscribing to `ContextClient`.
The subscriber of the `ContextClient` is responsible for unsubscribing

ref: [RxJs](https://github.com/ReactiveX/rxjs/blob/afac3d574323333572987e043adcd0f8d4cff546/src/internal/Observable.ts#L334)